### PR TITLE
feat: Add metrics-proxy rock

### PR DESCRIPTION
**Pull request by IbraAoad**
- [x] I have signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] I am following the [CONTRIBUTING](https://github.com/canonical/oci-factory/blob/main/CONTRIBUTING.md) guidelines

*Ping the @canonical/rocks team.*

---

## Description
This PR adds the metrics-proxy ROCK to the OCI factory.

### Related issues
#268 

---

*Picture of a cool rock:*
cool-rock

### CODE REVIEW

**Comment by IbraAoad** [comment]
@cjdcordeiro Do you mind taking a look at this as well, thanks! :bowing_man:

**Review by cjdcordeiro — APPROVED** [review]
It lgtm. Thanks. Since it's the 1st PR, I'd also like another colleague to have a 2nd look before merging

FYI @clay-lake @zhijie-yang

**Review by zhijie-yang — APPROVED** [review]
Looks nice to me, thanks! A nitpick for future updates: Please leave a blank line at the end of each file.

File: oci/metrics-proxy/image.yaml
Comment (clay-lake) [inline oci/metrics-proxy/image.yaml:5]: Is it worth adding the [0.1.0](https://github.com/canonical/metrics-proxy-rock/tree/main/0.1.0) rock as well?

**Review by clay-lake — CHANGES_REQUESTED** [review]
Would it be possible to add a newline at every EOF? I also added a comment about the prior version.

**Comment by clay-lake** [comment]
Quick note: we are waiting for #308 to be merged in. Should be completed shortly.

**Pull request merged**
